Improve the Helm readme

joestringer opened the pull request:
Improve the Helm readme by:
* Adding additional flavour text so that the page stands alone as a representation for the project
* Ensuring that the descriptions from the values.yaml are propagated into the values in the readme
* Extra metadata that external sites (such as artifacthub.io) can understand and use to render better pages referencing our helm charts
* Remove unnecessary options ('kvstore', 'wellKnownIdentities')

For a preview, see the readme in this branch:
https://github.com/joestringer/cilium/tree/submit/helm-improvements/install/kubernetes/cilium

Also nominating for backport to allow these to propagate to external sites when we spin the next release.

joestringer committed:
helm: Improve Cilium helm charts

Expand the charts to include other common fields so that external
systems can pull & use icons, tags, k8s version support, etc.

Furthermore, flesh out the readme so it renders better.

While we're at it, add an SVG without the cilium word so that it's more
likely to render correctly in a square on external sites.

Signed-off-by: Joe Stringer <joe@cilium.io>

joestringer committed:
helm: Remove 'kvstore' option.

This was not referenced anywhere else, remove it.

Signed-off-by: Joe Stringer <joe@cilium.io>

joestringer committed:
helm: Remove wellKnownIdentities

This option was too specific and doesn't make sense to expose as a
user-visible flag in the helm charts. Remove it.

Signed-off-by: Joe Stringer <joe@cilium.io>

joestringer committed:
helm: Add LICENSE to charts

This should allow external sites to better represent Cilium in artifacts
generated from the Helm charts.

Signed-off-by: Joe Stringer <joe@cilium.io>

joestringer committed:
helm: Add artifacthub CRD annotations.

Artifact Hub suggests that if we add such annotations to our charts,
then they will show up in the Artifact Hub UI somewhere:

https://artifacthub.io/docs/topics/annotations/helm/

Signed-off-by: Joe Stringer <joe@cilium.io>

joestringer committed:
helm: Fix autogeneration of helm value descriptions

These mostly needed '--' at the start for the helm-docs container to
pick up the descriptions and render them into the README.md.

While we're at it, Remove stuttering via the old go style of typing the
variable at the start of the sentence, use full sentences, and remove
any comments that were named exactly the same as the variable.

Signed-off-by: Joe Stringer <joe@cilium.io>

sayboras commented on install/kubernetes/cilium/README.md.gotmpl:
````diff
@@ -4,15 +4,51 @@
+Or, for a quick install with the default configuration:
+
+```
+$ helm repo add cilium https://helm.cilium.io/
````
most of the guide that I came across for helm chart including `helm repo update` after add, maybe just in case repo is added but there is new version :thinking:

joestringer replied:
Hmm, I hadn't seen that before. It's not part of our main instructions either. Personally I don't see a strong motivation to do so, as it should have only a pretty minor impact unless you just happen to be trying to install during one of 3-4 times per year when we make a brand new feature release.

sayboras commented on install/kubernetes/cilium/Chart.yaml:
```diff
@@ -1,9 +1,53 @@
 apiVersion: v2
 name: cilium
+displayName: Cilium
```
do you have a link to Chart.yaml schema?. I taked a look here https://helm.sh/docs/topics/charts/, but `displayName` and `links` are not mentioned.

joestringer replied:
I'm not sure if those ones are part of some default main schema but I found them while looking through the artifact hub documentation. Hopefully most tooling is willing to overlook fields they don't understand.

sayboras replied:
oh right :100:, all good :)

sayboras commented on install/kubernetes/cilium/Chart.yaml:
```diff
@@ -1,9 +1,53 @@
 appVersion: 1.9.90
-description: Helm chart for Cilium
+kubeVersion: ">= 1.12"
+icon: https://raw.githubusercontent.com/cilium/cilium/master/Documentation/images/logo-solo.svg
```
seems not found for me :thinking:

joestringer replied:
This one is only in the PR, so we would have to merge it first before it will work :)

michi-covalent replied:
looks like the image is a part of this pr, so i guess it'll magically appear when it gets merged.

sayboras replied:
oops my bad :face_with_thermometer:

joestringer replied:
That's what I'm hoping at least :) I was following similar examples from other projects on artifact hub, I hope this is the step we need to make the logo show up over there.

sayboras reviewed (approved):
very nice :100: :heavy_check_mark:

michi-covalent reviewed (approved):
lgtm 👍 should be good to merge once the feedback from @sayboras gets addressed 💯

kaworu reviewed (approved):
Great patch, LGTM!

christarazi commented on install/kubernetes/cilium/Chart.yaml:
```diff
@@ -20,3 +20,34 @@ sources:
   - name: eBPF.io
     url: https://ebpf.io/
+annotations:
+  artifacthub.io/crds: |
```
What about CiliumNode, CiliumIdentity, CiliumEndpoint?

joestringer replied:
I didn't bother with those since they're managed directly by Cilium, and other than providing a way to introspect cluster state, users won't be interacting with them (unlike the ones that are here, which are explicitly configured by the user).

:shrug: maybe we should just list all of them.

christarazi replied:
Yeah that makes sense. There is one specific workflow for CiliumNode where we ask the user in ENI mode if they want to change the interface offset (`first-interface-index`) for their ENI setup, to do it by modifying the CN resource. https://docs.cilium.io/en/v1.9/concepts/networking/ipam/eni/#ip-allocation

At the very least, we should "expose" CN. But yeah, I don't mind either way if we include all or keep it as is + CN.

christarazi reviewed (approved):
LGTM. One question below about CRDs.

joestringer committed:
helm: Fix documentation URL for v1.10

v1.10 will need more than the first 3 digits of the version semver, it
will need the full $major.$minor. Fix this up.

Signed-off-by: Joe Stringer <joe@cilium.io>

fristonio commented on install/kubernetes/cilium/README.md:
```diff
@@ -2,355 +2,320 @@
 
 ![Version: 1.9.90](https://img.shields.io/badge/Version-1.9.90-informational?style=flat-square) ![AppVersion: 1.9.90](https://img.shields.io/badge/AppVersion-1.9.90-informational?style=flat-square)
```
I was wondering what these versions represent? :confused: 
Shouldn't these be `1.9.0`?

joestringer replied:
In the tree during v1.10 development, we use the version number v1.9.90 today to denote "a version that may be much greater than v1.9.x but still a bit less than v1.10.0".

fristonio reviewed (approved):
Looks great, thanks for this :pray: :100: :rocket:

aanm commented on install/kubernetes/cilium/Chart.yaml:
```diff
@@ -1,9 +1,77 @@
 appVersion: 1.9.90
-description: Helm chart for Cilium
+kubeVersion: ">= 1.12.0-0"
+icon: https://raw.githubusercontent.com/cilium/cilium/master/Documentation/images/logo-solo.svg
```
@joestringer I assume we should make this logo dependent on the branch we are? i.e. if we backport this to 1.9 we need to change `cilium/master` to `cilium/v1.9`?

joestringer replied:
Addressed with the top commit.

joestringer committed:
helm: Link logo via CDN based on git branch

Signed-off-by: Joe Stringer <joe@cilium.io>
